src/vm/database/sqlite.rs

```diff
@@ -0,0 +1,305 @@
+use rusqlite::{Connection, OptionalExtension, NO_PARAMS, Row};
+use rusqlite::types::ToSql;
+
+use vm::contracts::Contract;
+use vm::database::{ContractDatabase};
+use vm::errors::{Error, ErrType, InterpreterResult as Result, IncomparableError};
+use vm::types::{Value, TypeSignature, TupleTypeSignature, AtomTypeIdentifier};
+
+
+pub struct SqliteContractDatabase {
+    conn: Option<Connection>,
+    save_point: u16
+}
+
+pub struct SqliteDataMap {
+    map_identifier: i64,
+    key_type: TypeSignature,
+    value_type: TypeSignature
+}
+
+impl SqliteContractDatabase {
+    pub fn initialize(filename: &str) -> Result<SqliteContractDatabase> {
+        let mut contract_db = SqliteContractDatabase::inner_open(filename)?;
+        contract_db.execute("CREATE TABLE IF NOT EXISTS maps_table
+                      (map_identifier INTEGER PRIMARY KEY AUTOINCREMENT,
+                       contract_name TEXT,
```

Comment:
Highly recommend adding the `NOT NULL` constraint to these columns.  Also, is the `contract_name` ultimately going to become the primary key -- i.e. the transaction ID that instantiated it?

Reply:
Yes, I'll add `NOT NULL` and `FOREIGN KEY` constraints to these tables as the schema finalizes.

No, I don't think `contract_name` can be the primary key -- a given contract may define multiple data maps.

Reply:
Gotcha. Maybe the primary key would be `(contract_name,map_identifier)` then?

Reply:
I think it's more efficient to leave the primary key as a single column -- this way, we can use just the `map_identifier` to denote which map a particular entry in the `data_table` corresponds to, rather than needing to use two columns there as well. We will, I'm sure, want to index on `contract_name` in this table, though.